data/virt_autotest/autoyast/host_15.xml.ep

```diff
@@ -0,0 +1,268 @@
+      % }
+    </global>
+    <loader_type>grub2</loader_type>
+  </bootloader>
+  <general>
+    <ask-list config:type="list"/>
+    <mode>
+      <confirm config:type="boolean">false</confirm>
+      <final_reboot config:type="boolean">true</final_reboot>
+    </mode>
+    <proposals config:type="list"/>
+    <signature-handling>
+      <accept_file_without_checksum config:type="boolean">true</accept_file_without_checksum>
+      <accept_non_trusted_gpg_key config:type="boolean">true</accept_non_trusted_gpg_key>
+      <accept_unknown_gpg_key config:type="boolean">true</accept_unknown_gpg_key>
+      <accept_unsigned_file config:type="boolean">true</accept_unsigned_file>
+      <accept_verification_failed config:type="boolean">true</accept_verification_failed>
+      <import_gpg_key config:type="boolean">true</import_gpg_key>
+    </signature-handling>
+    <storage>
+      <start_multipath config:type="boolean">false</start_multipath>
+    </storage>
+  </general>
+  <kdump>
+    <add_crash_kernel config:type="boolean">true</add_crash_kernel>
+    <crash_kernel>201M,high</crash_kernel>
+    <crash_xen_kernel>201M\&lt;4G</crash_xen_kernel>
+  </kdump>
+  <keyboard>
+    <keymap>english-us</keymap>
+  </keyboard>
+  <language>
+    <language>en_US</language>
+    <languages/>
+  </language>
+  <login_settings/>
+  <networking>
+    <keep_install_network config:type="boolean">true</keep_install_network>
+    <backend>wicked</backend>
+    <dns>
+      <dhcp_hostname config:type="boolean">true</dhcp_hostname>
+      <nameservers config:type="list">
+        <nameserver>10.162.0.1</nameserver>
+<!--gonzo and a another machine can not get nameserver from local dns server after autoyast installation. Put a static nameserver option above. If nameserver is retrived from dhcp they will be merged. Otherwise use the static -->
+      </nameservers>
+      <resolv_conf_policy>auto</resolv_conf_policy>
+    </dns>
+    <virt_bridge_proposal config:type="boolean">true</virt_bridge_proposal>
+  </networking>
+  <ntp-client>
+    <ntp_policy>auto</ntp_policy>
+    <ntp_servers config:type="list">
+      <ntp_server>
+        <address>europe.pool.ntp.org</address>
+        <iburst config:type="boolean">true</iburst>
+        <offline config:type="boolean">false</offline>
+      </ntp_server>
+    </ntp_servers>
+    <ntp_sync>systemd</ntp_sync>
+  </ntp-client>
+  <partitioning config:type="list">
+  % my $wwn = {'quinn' => 'wwn-0x5000c50099db2117', 'kermit-1' => 'wwn-0x500a075119406ab6', 'gonzo-1' => 'wwn-0x500a075119406aa6', 'fozzie-1' => 'wwn-0x55cd2e414f1f16f1', 'scooter-1' => 'wwn-0x55cd2e414f1760e2', 'amd-zen3-gpu-sut1-1' => 'wwn-0x500a075133755d4b'};
+  % my $hostname = (split(/\./, $get_var->("SUT_IP")))[0];
+  % my $device_id = defined($wwn->{$hostname}) ? '/dev/disk/by-id/' . $wwn->{$hostname} : ''; 
+    <drive>
+      <device><%= $device_id %></device>
```

Comment:
Would you please explain the purpose/benefit of setting $wwn? In OSD, we have more than 6 SUTs. Why 6 SUTs are configured here? Miss some SUT or only need to set them?

Besides, if there can be other methods/workarounds to achieve the given purpose,  it will be good to avoid giving hardware specific info in autoyast. Otherwise, once we have new machines, we will always have to update autoyast. 
If it is not avoidable, maybe moving such hardware info (wwn) to worker.ini, then extract the settings by get_var and use it here, will be better?

Reply:
> Would you please explain the purpose/benefit of setting $wwn? 
In OSD, we have more than 6 SUTs. Why 6 SUTs are configured here? Miss some SUT or only need to set them?
>
> Besides, if there can be other methods/workarounds to achieve the given purpose, it will be good to avoid giving hardware specific info in autoyast. 

They are using lun from a array storage (maybe two) .  I have to use persistent device id to make sure OS is installed on first lun.  It's not bootable with OS on other luns.  see more in the ticket.

Let me know the name of missing SUTs. 

A question:  After your tests install a sles build and reboot. How you do know if the installed sles is the right build rather then you boot a previously installed os from the second disk.  

Otherwise, once we have new machines, we will always have to update autoyast. If it is not avoidable, maybe moving such hardware info (wwn) to worker.ini, then extract the settings by get_var and use it here, will be better?

If the new machine uses lun as block device you have to update autoyast.  
Lun is not a hardware of machine.
Putting them in autoyast make people understand their  purpose easily as it's used in here. 
People like to edit one file more than edit 6 files.

Reply:
All machines in https://gitlab.suse.de/openqa/salt-pillars-openqa/-/blob/master/openqa/workerconf.sls with worker class `64bit-ipmi` are what we use on OSD :) .

I am not very convinced by the reasons of not putting in worker.ini. But I do not insist on this. Just share my idea here.
- Still 1 file needs to be edited, workerconf.sls which is automatically deployed by salt to worker.ini . 
- LUN id seems closely related with each SUT, fine enough to put in worker.ini .

@Julie-CAO How do you handle in TW autoyast to let installation done on first disk? Will it be good to unify the method for this?

Regarding how we confirm the correct os is booted from, we have been always installing on first disk by selecting first disk during interactive os installation. So there is no os at all on second disk. No such trouble to worry :)

Reply:
I agree to put these wwns for particular SUTs in workers.ini in salt. They are more dependent on machine itself than the autoyast profile. Given the hostname of a SUT is changed in salt, the autoyast have to be changed as well. Or if a new machine is added in salt, the autoyast has be changed well. Putting them in workers.ini and add an setting will save future maintaince effort.